packages/block-editor/src/components/block-lock/modal.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+import {
+	Button,
+	CheckboxControl,
+	Flex,
+	FlexItem,
+	Icon,
+	Modal,
+} from '@wordpress/components';
+import { dragHandle, trash } from '@wordpress/icons';
+import { useInstanceId } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import useBlockDisplayInformation from '../use-block-display-information';
+import { store as blockEditorStore } from '../../store';
+
+export default function BlockLockModal( { clientId, onClose } ) {
+	const [ lock, setLock ] = useState( { move: false, remove: false } );
+	const { canMove, canRemove } = useSelect(
+		( select ) => {
+			const {
+				canMoveBlock,
+				canRemoveBlock,
+				getBlockRootClientId,
+			} = select( blockEditorStore );
+			const rootClientId = getBlockRootClientId( clientId );
+
+			return {
+				canMove: canMoveBlock( clientId, rootClientId ),
+				canRemove: canRemoveBlock( clientId, rootClientId ),
+			};
+		},
+		[ clientId ]
+	);
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+	const blockInformation = useBlockDisplayInformation( clientId );
+	const instanceId = useInstanceId(
+		BlockLockModal,
+		'block-editor-block-lock-modal__options-title'
+	);
+
+	useEffect( () => {
+		setLock( {
+			move: ! canMove,
+			remove: ! canRemove,
+		} );
+	}, [ canMove, canRemove ] );
+
+	const isAllChecked = Object.values( lock ).every( Boolean );
+
+	let ariaChecked;
+	if ( isAllChecked ) {
+		ariaChecked = 'true';
+	} else if ( Object.values( lock ).some( Boolean ) ) {
+		ariaChecked = 'mixed';
+	} else {
+		ariaChecked = 'false';
+	}
+
+	return (
+		<Modal
+			title={ sprintf(
+				/* translators: %s: Name of the block. */
+				__( 'Lock %s' ),
+				blockInformation.title
+			) }
+			overlayClassName="block-editor-block-lock-modal"
+			closeLabel={ __( 'Close' ) }
+			onRequestClose={ onClose }
+		>
+			<form
+				onSubmit={ ( event ) => {
+					event.preventDefault();
+					updateBlockAttributes( [ clientId ], { lock } );
+					onClose();
+				} }
+			>
+				<p>
+					{ __(
+						'Choose specific attributes to restrict or lock all available options.'
+					) }
+				</p>
+				<div
+					role="group"
+					aria-labelledby={ instanceId }
+					className="block-editor-block-lock-modal__options"
+				>
+					<CheckboxControl
+						className="block-editor-block-lock-modal__options-title"
+						label={
+							<span id={ instanceId }>{ __( 'Lock all' ) }</span>
+						}
+						checked={ isAllChecked }
+						aria-checked={ ariaChecked }
```

Comment:
Can you switch to using this?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#attr-indeterminate

Seems like aria-checked is not taking effect due to the missing checkbox role.

Reply:
Looks like you can pass additional props but might be nice to have the support.

https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/checkbox-control/index.js#L54

Reply:
According to MDN:

> This is set using the [HTMLInputElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement) object's indeterminate property via JavaScript (it cannot be set using an HTML attribute).

So passing it as prop/attribute won't work.

Reply:
Oh, so this has to be queried by ID or class directly? Unbelievable. 👎 What happens if you add `role="checkbox"`?

Reply:
It's possible to support this feature in custom components. Do you mind creating a new issue for `indeterminate` support?

> What happens if you add role="checkbox"?

I assumed inputs with type="checkbox" would have this role automatically assigned.

Reply:
@Mamaduka I will open an issue. For now, how hard would it be to convert this to radio buttons?

Reply:
Thanks for the ping. We're definitely going to take a look and report back